### PR TITLE
Simplify macOS build instruction

### PR DESCRIPTION
**Pull request by TachikakaMin**
cc @svekars @carljparker

### CODE REVIEW

**Comment by TachikakaMin** [comment]
@msaroufim @soumith

Hi all, 

Maybe you can check this PR since you all have changed this README MacOS compile part (:

File: README.md [hunk 1]
```diff
@@ -249,11 +249,20 @@ This is caused by `ld` from the Conda environment shadowing the system `ld`. You
...
 MACOSX_DEPLOYMENT_TARGET=10.9 CC=clang CXX=clang++ python setup.py develop
 ```
 
+For default macOS configuration, you can just use 
+
+```bash
+export CMAKE_PREFIX_PATH=${CONDA_PREFIX:-"$(dirname $(which conda))/../"}
+MACOSX_DEPLOYMENT_TARGET=10.9 CC=gcc CXX=g++ python setup.py develop
+```
+
```
Comment (malfet) [inline README.md:265]: This is how I usually build PyTorch on MacOS for the last 3 years. I.e. if one builds using conda environment, `CMAKE_PREFIX_PATH` should already be defined. If not, no reason to tweak the variables. But I think it's important to call out that `python3` should be used, as on Intel Mac `python` still points to default Python 2.7 runtime, which is not supported for the last 3 releases of PyTorch
```suggestion
```bash
python3 setup.py develop
```

For default macOS configuration, you can just use 

```bash
export CMAKE_PREFIX_PATH=${CONDA_PREFIX:-"$(dirname $(which conda))/../"}
MACOSX_DEPLOYMENT_TARGET=10.9 CC=gcc CXX=g++ python setup.py develop
```

```

**Review by malfet — CHANGES_REQUESTED** [review]
Thank you for your interest in improving PyTorch documentation.

The file you are referencing is only relevant if one tests PyTorch for iOS/Android devices, while for a vanilla MacOS build it shouldn't make much difference (Also, if user builds  `clang` from source instead of using one provided by XCode/Developer tools, they should be pretty comfortable about the distinction between the two.)

Do you mind adjusting your change to include a very simple instruction that works for 99% of the users (i.e. when one builds locally, by default there are no need specifying deployment targets nor CC/CXX one supposed to use)

File: README.md (hunk 1 above)
Comment (TachikakaMin) [reply README.md:265]: You are right. 

Actually, the bug is in the file I'm referencing. AppleClang and Clang don't have --whole-archive option.

**Comment by TachikakaMin** [comment]
The bug is in the file I'm referencing. AppleClang and Clang don't have --whole-archive option.

When I first compile code on my M1 computer, I thought it because clang(or AppleClang) in macos doesn't support  "--whole-archive" so I want to change the readme file.

File: README.md (hunk 1 above)
Comment (malfet) [reply README.md]: `--whole-archive` is a linker option(notice, that flag is passed as `-Wl,--whole-archive`), and `-force_load` is indeed available on MacOS platform (below is the quote from `man ld`):
>      -force_load path_to_archive
>             Loads all members of the specified static archive library.  Note:
>             -all_load forces all members of all archives to be loaded.  This
>             option allows you to target a specific archive.

If you want to fix this, please do not hesitate to open a new PR and change condition from `if(CMAKE_CXX_COMPILER_ID to `if(APPLE)` or something like that

**Review by malfet — CHANGES_REQUESTED** [review]
Please do not conflate two changes into one: change `README.md` in this PR, but then update the condition in another one.

Also please note, that  current change as it stands is wrong, since if this code is executed on linux will fail, where linker indeed supports `--whole-archive` rather than `-force_load`.

File: README.md (hunk 1 above)
Comment (TachikakaMin) [reply README.md]: That's true. I will open a new PR.

**Comment by TachikakaMin** [comment]
> Please do not conflate two changes into one: change `README.md` in this PR, but then update the condition in another one.
> 
> Also please note, that current change as it stands is wrong, since if this code is executed on linux will fail, where linker indeed supports `--whole-archive` rather than `-force_load`.

Yes, you are right.

File: README.md (hunk 1 above)
Comment (TachikakaMin) [reply README.md]: New PR is here: https://github.com/pytorch/pytorch/pull/91736

File: README.md [hunk 2]
```diff
@@ -250,10 +250,7 @@ This is caused by `ld` from the Conda environment shadowing the system `ld`. You
 **On macOS**
 
 ```bash
-export CMAKE_PREFIX_PATH=${CONDA_PREFIX:-"$(dirname $(which conda))/../"}
-MACOSX_DEPLOYMENT_TARGET=10.9 CC=clang CXX=clang++ python setup.py develop
-```
-
+python3 setup.py develop
```
Comment (malfet) [inline README.md:253]:
```suggestion
python3 setup.py develop
```
```

**Review by malfet — CHANGES_REQUESTED** [review]
Almost, LGTM but you forgot the closing backticks block, aren't you (and I can't propose this as a suggestion

**Review by malfet — APPROVED** [review]
Well, actually I can, looks good to me now :)

**Pull request closed without merging**